openpype/hosts/mrv2/startup/mrv2_openpype_plugin.py

```diff
@@ -0,0 +1,84 @@
+from mrv2 import plugin, cmd
+import contextlib
+
+timer = None
+
+
+def separator():
+    # Do nothing function, temporary until mrv2 exposes separator functionality
+    pass
+
+
+class MyPlugin(plugin.Plugin):
+    def on_create(self):
+        print("Create..")
+        from openpype.tools.utils import host_tools
+        with qt_with_mrv_update():
+            host_tools.show_publisher(tab="create")
+
+    def on_load(self):
+        print("Load..")
+        from openpype.tools.utils import host_tools
+        with qt_with_mrv_update():
+            host_tools.show_loader(use_context=True)
+
+    def on_publish(self):
+        print("Publish..")
+        from openpype.tools.utils import host_tools
+        with qt_with_mrv_update():
+            host_tools.show_publisher(tab="publish")
+
+    def on_manage(self):
+        print("Manage..")
+        from openpype.tools.utils import host_tools
+        with qt_with_mrv_update():
+            host_tools.show_scene_inventory()
+
+    def on_library(self):
+        print("Library..")
+        from openpype.tools.utils import host_tools
+        with qt_with_mrv_update():
+            host_tools.show_library_loader()
+
+    def on_workfiles(self):
+        print("Workfiles..")
+        from openpype.tools.utils import host_tools
+        with qt_with_mrv_update():
+            host_tools.show_workfiles()
+
+    def menus(self):
+        top = "OpenPype"
+        return {
+            f"{top}/Create...": self.on_create,
+            f"{top}/Load...": self.on_load,
+            f"{top}/Publish...": self.on_publish,
+            f"{top}/Manage...": self.on_manage,
+            f"{top}/Library...": self.on_library,
+            f"{top}/-------": separator,
+            f"{top}/Workfiles...": self.on_workfiles,
+        }
+
+
+@contextlib.contextmanager
+def qt_with_mrv_update():
+    from qtpy import QtCore
+    from openpype.tools.utils import qt_app_context
+    with qt_app_context() as app:
+        global timer
+        if timer is None:
+            timer = QtCore.QTimer(parent=app)
+            timer.setInterval(0)
+            timer.timeout.connect(cmd.update)
+            timer.start()
```

Comment:
@iLLiCiTiT do you know a better approach to this?

Would this logic still work if I close the last window, e.g. close the Loader UI - then I assume the `QApplication` exists because it quits on last window close - which I what I'd like since whenever the Qt windows are closed again I want no strain on MRV2 to persist.

But the `QApplication` instance would still exist? The timer still reports to be active `timer.isActive()` returns True. But if I attach it to a custom function that also prints it seems the second time around the prints have stopped.

Anyone know how to ensure the behavior I'm looking for?
Do I also need to re-exec the QApplication instance if it has been closed down due to no more active windows?

Reply:
I just read it in hurry, so I'm sorry if I missed something.

To avoid closing of QApplication call on it's creation this:
```python
app = QtWidgets.QApplication.instance()
app.setQuitOnLastWindowClosed(False)
```

Re-execution can cause glitches because the app thinks it's closing -> e.g. fonts are unloaded etc.

If you can't use `setQuitOnLastWindowClosed` you can look at `QtWidgets.QApplication.closingDown()`.

Reply:
I would've actually preferred the QApplication to spin down completely and close - if possible - since it'd mean that the timer would always need to keep running to make sure `mrv` updates as the QApplication execution loop runs. So basically the speed of MRV2 is then dependent on the QApplication QTimer callbacks but I'd rather have it that as soon as I close all Qt applications that we basically 'lose all' to ensure we get top speed from MRV2 again.

Anyway, will take another look.

Reply:
Tweaked the code - this seemed to work as I wanted in my test runs.

Reply:
It seems MRV2 isn't all too happy about having Qt event loop running next to its own. https://github.com/ggarra13/mrv2/issues/130

We might need to look into other options. :)

Reply:
It seems magically solved by just not calling `QApplication.exec_()` - the UIs run fine, not sure why. I recall seeing similar behavior in Blender in the past too. Anyway, for now I've just let it be since it just worked for me as is.